src/Servers/Kestrel/Core/test/HeartbeatTests.cs

```diff
@@ -56,7 +57,7 @@ public async Task HeartbeatTakingLongerThanIntervalIsLoggedAsError()
             await blockedHeartbeatTask.DefaultTimeout();
 
             heartbeatHandler.Verify(h => h.OnHeartbeat(now), Times.Once());
-            kestrelTrace.Verify(t => t.HeartbeatSlow(TimeSpan.FromSeconds(2), Heartbeat.Interval, now), Times.Once());
+            Assert.Equal($"As of\"{now}\", the heartbeat has been running for \"{heartbeatDuration}\" which is longer than \"{Heartbeat.Interval}\". This could be caused by thread pool starvation.", kestrelTrace.Logger.Messages.Single(message => message.LogLevel == LogLevel.Warning).Message);
```

Comment:
```suggestion
            Assert.Equal($"As of \"{now.ToString(CultureInfo.InvariantCulture)}\", the heartbeat has been running for \"{heartbeatDuration.ToString("c", CultureInfo.InvariantCulture)}\" which is longer than \"{Heartbeat.Interval.ToString("c", CultureInfo.InvariantCulture)}\". This could be caused by thread pool starvation.", kestrelTrace.Logger.Messages.Single(message => message.LogLevel == LogLevel.Warning).Message);
```